### ISSUE TITLE
Missing array element statement (due to Dexpler?)

### ISSUE THREAD

**Issue by MarcMil** [Sable/soot#520]
Running
java -jar ../soot-trunk.jar --allow-phantom-refs -f J -src-prec apk -cp /home/marc/adt-bundle-linux-x86_64-20140702/sdk/platforms/android-18/android.jar -process-dir /home/marc/com.antivirus.apk/Imported.apk

yields in sootOutput/a.a.a.a.a.a.jimple
```
    static void <clinit>()
……

        $r0[64] = -1;

        $r0[67] = 2;
```

There should be a
```
$r0[66] = 1;
```
in between.

Corresponding dex code:
```
10f930: 1301 4200                              |012a: const/16 v1, #int 66 // #42
10f934: 1212                                   |012c: const/4 v2, #int 1 // #1
10f936: 4f02 0001                              |012d: aput-byte v2, v0, v1
```

Also, later on, quite similar, a
```
$r0[57] = 61;

10f8f8: 1301 3900                              |010e: const/16 v1, #int 57 // #39
10f8fc: 1302 3d00                              |0110: const/16 v2, #int 61 // #3d
10f900: 4f02 0001                              |0112: aput-byte v2, v0, v1
```

and
```
$r0[90] = 25;

10fa40: 1301 5a00                              |01b2: const/16 v1, #int 90 // #5a
10fa44: 1302 1900                              |01b4: const/16 v2, #int 25 // #19
10fa48: 4f02 0001                              |01b6: aput-byte v2, v0, v1
```

is missing.
I have no clue why this does not appear in the Jimple code...

The APK is the com.antivirus.apk file.
https://www.dropbox.com/s/25wgl1qt2cso9co/Imported.apk?dl=0

**Comment by Alexandre-Bartel** [comment]
Fixed in e523e502f51b4eebedb95ed30ccf0661de7ed267. FYI the problem was in new optimization code which optimized a bit too much...

**Comment by MarcMil** [comment]
Thanks a lot, Alexandre!
